File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int low = 0;
        long sum = 0;
        int high = numbers.length-1;
        while(low <= high){
             sum = numbers[low] + numbers[high];
            if(sum == target){
                return new int [] {low+1,high+1};
            }
            else if(sum > target){
                high--;
            }
            else{
                low++;
            }
        }
        return new int[] {low+1,high+1};
        
    }
}

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        // using 2 pointer patter
        int left = 0;
        int right = numbers.size() - 1;
        vector <int> ans(2);
        while(left < right){
            int sum = numbers[left] + numbers[right];
            if(sum == target) {
                ans[0] = left + 1;
                ans[1] = right + 1;
                return ans;
            }
            else if(sum > target) right--;
            else{
                left++;
            }

        }
        return ans;
        
    }
};